kctf-operator/pkg/apis/kctf/v1alpha1/challenge_types.go

```diff
@@ -0,0 +1,130 @@
+// This file is responsible for generating CRD (Custom Resource Definition)
+// kubebuilder might be used to set: default values, optional fields and etc
+// +kubebuilder:validation:Required
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Network specifications for the service
+type NetworkSpec struct {
+
+	// +kubebuilder:default:=true
+	Public bool `json:"public,omitempty"`
+
+	// +kubebuilder:default:=false
+	Dns bool `json:"dns,omitempty"`
+
+	// By default, one port is set with default values
+	// +kubebuilder:validation:Optional
+	Ports []corev1.ServicePort `json:"ports,omitempty"`
```

Comment:
I tested it and setting `protocol: "HTTP"` does work; but it doesn't work if you not specifying `port:` value. And `port` doesn't make sense with HTTP because it will be 80/443 anyway.
So maybe we better introduce our own object for that instead of using corev1.ServicePort?
It would be the same as corev1.ServicePort but it would have `port` as optional and allow `"HTTP"` value in protocol. Because using the standard object might confuse users (they can be surprised by usage of "HTTP" inside corev1.ServicePort)

Reply:
+1

Reply:
I didn't know that it's not supported by k8s. How do we pass currently a port protocol "HTTP" then?

Reply:
We don't :) It's a change that comes with CRD migration.

What we have to do is:
* if the port is of type TCP, we create a LoadBalancer (just pass the values to the Service)
* if the port is of type HTTP we create an Ingress

Reply:
Ok, that makes sense. I will create then a new type with only the fields that we discussed (port, targetPort, name and protocol) and then create an Ingress if the protocol is HTTP. Btw, where is the Ingress created so I know which fields need to be configured in the Ingress?

Reply:
`INGRESS_TEMPLATE` in https://github.com/google/kctf/blob/alpha/base/Makefile#L164

Reply:
And where is it used? I was looking at the specs of Service but didn't find it easily.